cmd/client_brownfield_cmd.go

```diff
@@ -0,0 +1,158 @@
+package cmd
+
+import (
+	"github.com/ingrammicro/cio/api/clientbrownfield"
+	"github.com/ingrammicro/cio/utils"
+	"github.com/ingrammicro/cio/utils/format"
+	"github.com/urfave/cli"
+)
+
+// WireUpBrownfieldCloudAccount prepares common resources to send request to Concerto API
+func WireUpBrownfieldCloudAccount(c *cli.Context) (ds *clientbrownfield.BrownfieldCloudAccountService, f format.Formatter) {
+
+	f = format.GetFormatter()
+
+	config, err := utils.GetConcertoConfig()
+	if err != nil {
+		f.PrintFatal("Couldn't wire up config", err)
+	}
+	hcs, err := utils.NewHTTPConcertoService(config)
+	if err != nil {
+		f.PrintFatal("Couldn't wire up concerto service", err)
+	}
+	ds, err = clientbrownfield.NewBrownfieldCloudAccountService(hcs)
+	if err != nil {
+		f.PrintFatal("Couldn't wire up Brownfield Cloud Account service", err)
+	}
+
+	return ds, f
+}
+
+// BrownfieldCloudAccountList subcommand function
+func BrownfieldCloudAccountList(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	svc, formatter := WireUpBrownfieldCloudAccount(c)
+
+	cloudAccounts, err := svc.ListBrownfieldCloudAccounts()
+	if err != nil {
+		formatter.PrintFatal("Couldn't receive cloud accounts data", err)
+	}
+
+	cloudProvidersMap := LoadCloudProvidersMapping(c)
+	for id, ca := range cloudAccounts {
+		cloudAccounts[id].CloudProviderName = cloudProvidersMap[ca.CloudProviderID]
+	}
+
+	if err = formatter.PrintList(cloudAccounts); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// BrownfieldCloudAccountShow subcommand function
+func BrownfieldCloudAccountShow(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	cloudAccountSvc, formatter := WireUpBrownfieldCloudAccount(c)
+
+	checkRequiredFlags(c, []string{"id"}, formatter)
+	cloudAccount, err := cloudAccountSvc.GetBrownfieldCloudAccount(c.String("id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't receive cloud account data", err)
+	}
+
+	cloudProvidersMap := LoadCloudProvidersMapping(c)
+	cloudAccount.CloudProviderName = cloudProvidersMap[cloudAccount.CloudProviderID]
+
+	if err = formatter.PrintItem(*cloudAccount); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// BrownfieldCloudAccountServersDiscover subcommand function
+func BrownfieldCloudAccountServersDiscover(c *cli.Context) error {
```

Comment:
From my understanding, discovery methods (server, vpn, floating ip, volume) should be able to wait for list methods before returning, right?
@pbanos Any directive? Maybe requesting for List<Resource> until IMCO resolves with value?...

Reply:
Cloud accounts have a last_discovered_at timestamp that is updated after a discover finishes. You should poll the discovered cloud account until that changes and then list the candidates

Reply:
Done